### ISSUE TITLE
Separate types for readonly Store (`Store`) and listen-only Event (`Event`)

### ISSUE THREAD

**Issue by igorkamyshev** [effector/effector#847]
## Proposal

1. Mapped stores should not be modified, we can show it on type level:

```ts
const $s1 = createStore(''); // StoreWritable<string>
const $s2 = $s1.map(s => s.length); // Store<number>

sample({
  clock: someEvent,
  target: $s2,
})
// TYPE ERROR
```

Same for derived events 

```ts
const derivedEvent = sample({ soure: $s, clock: event }); // Event<string>

sample({
  clock: someEvent,
  target: derivedEvent,
})
// TYPE ERROR
```

**Comment by sergeysova** [comment]
`Store` renamed into `StoreWritable`, but `ReadonlyStore` renamed into `Store`.

`Event` renamed into `EventCallable`, but `ReadonlyEvent` renamed into `Event`.

**Comment by YanLobat** [comment]
Resolved via #966
